msal4j-sdk/src/main/java/com/microsoft/aad/msal4j/CacheRefreshReason.java

```diff
@@ -0,0 +1,31 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.microsoft.aad.msal4j;
+
+/**
+ * List of possible reasons the tokens in an {@link IAuthenticationResult} were refreshed.
+ */
+public enum CacheRefreshReason {
```

Comment:
You are missing a case here when "Claims" are specified. When this happens, MSAL has to bypass the token cache and to ESTS. 

But not for client capabilities. Client capabilities are internally still claims. But we should not bypass the cache.

Reply:
MSAL.NET bundles Force Refresh + Claims https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/blob/fe3c8eacfb013b73404b3295ec5a36b3a60b3ed8/src/client/Microsoft.Identity.Client/Cache/CacheRefreshReason.cs#L18

Reply:
Still open comment :)